go/vt/vttablet/sandboxconn/sandboxconn.go

```diff
@@ -127,8 +128,10 @@ func (sbc *SandboxConn) getError() error {
 		sbc.MustFailCodes[code] = count - 1
 		return vterrors.New(code, fmt.Sprintf("%v error", code))
 	}
-	if sbc.ShardErr != nil {
-		return sbc.ShardErr
+	if sbc.EphemeralShardErr != nil {
+		err := sbc.EphemeralShardErr
+		sbc.EphemeralShardErr = nil
+		return err
```

Comment:
could use `sbc.MustFailCodes` with count 1 in place of EphemeralShardErr

Reply:
not really - I need a specific `SQLError` here, and `MustFailCodes` only supports `vterrors`, right?